nodePackages.scuttlebot won't build on Darwin

## Issue description

```
> level@4.0.0 postinstall /nix/store/ad5mdlvh9jdy2mqp04fhqiaj0af4gnpa-node-scuttlebot-13.2.2/lib/node_modules/scuttlebot/node_modules/level
> opencollective-postinstall || exit 0

sh: opencollective-postinstall: command not found

> leveldown@4.0.1 install /nix/store/ad5mdlvh9jdy2mqp04fhqiaj0af4gnpa-node-scuttlebot-13.2.2/lib/node_modules/scuttlebot/node_modules/leveldown
> prebuild-install || node-gyp rebuild

sh: prebuild-install: command not found
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/nix/store/70dmn2lwih4m3jryvf74j8rsp33kpzkd-nodejs-10.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 18.2.0
gyp ERR! command "/nix/store/70dmn2lwih4m3jryvf74j8rsp33kpzkd-nodejs-10.15.0/bin/node" "/nix/store/70dmn2lwih4m3jryvf74j8rsp33kpzkd-nodejs-10.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /nix/store/ad5mdlvh9jdy2mqp04fhqiaj0af4gnpa-node-scuttlebot-13.2.2/lib/node_modules/scuttlebot/node_modules/leveldown
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! leveldown@4.0.1 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the leveldown@4.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /private/tmp/nix-build-node-scuttlebot-13.2.2.drv-0/.npm/_logs/2019-03-13T20_49_10_219Z-debug.log
note: keeping build directory '/private/tmp/nix-build-node-scuttlebot-13.2.2.drv-0'
builder for '/nix/store/2jy1rhkygf7xab624ryrz0ynkrdbr0w6-node-scuttlebot-13.2.2.drv' failed with exit code 1
error: build of '/nix/store/2jy1rhkygf7xab624ryrz0ynkrdbr0w6-node-scuttlebot-13.2.2.drv' failed

```

--- a/pkgs/development/web/nodejs/nodejs.nix
+++ b/pkgs/development/web/nodejs/nodejs.nix
@@ -72,7 +72,6 @@ in
 
     postPatch = ''
       patchShebangs .
-      sed -i 's/raise.*No Xcode or CLT version detected.*/version = "7.0.0"/' tools/gyp/pylib/gyp/xcode_emulation.py
 
       # fix tests
       for a in test/parallel/test-child-process-env.js \
@@ -84,6 +83,7 @@ in
           --replace "/usr/bin/env" "${coreutils}/bin/env"
       done
     '' + optionalString stdenv.isDarwin ''
+      sed -i 's/raise.*No Xcode or CLT version detected.*/version = "7.0.0"/' tools/gyp/pylib/gyp/xcode_emulation.py
       sed -i -e "s|tr1/type_traits|type_traits|g" \
              -e "s|std::tr1|std|" src/util.h
     '';
@@ -101,6 +101,8 @@ in
 
       # install the missing headers for node-gyp
       cp -r ${concatStringsSep " " copyLibHeaders} $out/include/node
+    '' + optionalString stdenv.isDarwin ''
+      sed -i 's/raise.*No Xcode or CLT version detected.*/version = "7.0.0"/' $out/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py
     '';
 
     passthru.updateScript = import ./update.nix {

--- a/pkgs/development/web/nodejs/nodejs.nix
+++ b/pkgs/development/web/nodejs/nodejs.nix
@@ -101,7 +101,7 @@ in
 
       # install the missing headers for node-gyp
       cp -r ${concatStringsSep " " copyLibHeaders} $out/include/node
-    '' + optionalString stdenv.isDarwin ''
+    '' + optionalString (stdenv.isDarwin && enableNpm) ''
       sed -i 's/raise.*No Xcode or CLT version detected.*/version = "7.0.0"/' $out/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py
     '';
 

